src/main/scala/mission/impossibl/bots/sink/WasteSink.scala

```diff
@@ -23,6 +23,8 @@ object WasteSink {
               state.garbage_level + packet.total_mass,
               state.garbage_packets.updated(packet_uuid, packet)))
           case ProcessGarbage(garbage_packet_id) => // simulates garbage processing
+            // TODO change garbage_packets to list of tuples
+            // Change garbage_packet_id to n, get first n packets and process them
             val garbage_packet = state.garbage_packets.get(garbage_packet_id)
```

Comment:
Imo ta funkcjonalnośc mogłaby działać tak:
- env wysyła process garbage z X kg
- sink przechowuje "stan" tego ile z pakietu przeprocesował i jak przeprocesował w całości pakiet, bo przyszło process 5, 4, 4 (czyli razem 13kg), a pierwszy pakiet ma 12 kg, to zdejmuje pierwszy pakiet z kolejki, przerzuca na kolejkę przetworzonych i updejute stan przetwarzania na 1kg 

Wtedy możemy mieć listę partii śmieci od śmieciarek, a w każdej partii listę tupli (ilość śmieci, source_id) i zdejmować po prostu po kolei i nie bawić się getami na mapie i optionalami :)

Reply:
Alternatywnie: prędkość przetwarzania to parametr source'a. Source przy ticku zegara losuje z dystrybucji o niewielkim rozstrzale ile przprocesuje

Reply:
Zmodyfikowałem pierwsze podejście, modyfikuje parametr całkowitej masy paczki śmieci. Gdy osiągnie on zero jest zdejmowany z listy. e6b036e9655213d19b1b2d54908658241a57eb79